experimental/packages/otlp-grpc-exporter-base/src/OTLPGRPCExporterNodeBase.ts

```diff
@@ -15,79 +15,105 @@
+    signalSpecificMetadata: Record<string, string>,
+    grpcName: string,
+    grpcPath: string,
+    serializer: ISerializer<ServiceRequest, ServiceResponse>
+  ) {
     super(config);
+    this._serializer = serializer;
     if (config.headers) {
       diag.warn('Headers cannot be set when using grpc');
     }
-    const headers = baggageUtils.parseKeyPairsIntoRecord(
+    const nonSignalSpecificMetadata = baggageUtils.parseKeyPairsIntoRecord(
       getEnv().OTEL_EXPORTER_OTLP_HEADERS
     );
-    this.metadata = config.metadata || new Metadata();
-    for (const [k, v] of Object.entries(headers)) {
-      this.metadata.set(k, v);
+    const rawMetadata = Object.assign(
+      {},
+      nonSignalSpecificMetadata,
+      signalSpecificMetadata
+    );
+
+    let credentialProvider = () => {
+      return configureCredentials(undefined, this.getUrlFromConfig(config));
+    };
+
+    if (config.credentials != null) {
+      const credentials = config.credentials;
+      credentialProvider = () => {
+        return credentials;
+      };
     }
-    this.compression = configureCompression(config.compression);
-  }
 
-  private _sendPromise(
-    objects: ExportItem[],
-    onSuccess: () => void,
-    onError: (error: OTLPExporterError) => void
-  ): void {
-    const promise = new Promise<void>((resolve, reject) => {
-      this._send(this, objects, resolve, reject);
-    }).then(onSuccess, onError);
+    // Ensure we don't modify the original.
+    const configMetadata = config.metadata?.clone();
+    const metadataProvider = () => {
+      const metadata = configMetadata ?? createEmptyMetadata();
+      for (const [key, value] of Object.entries(rawMetadata)) {
+        // only override with env var data if the key has no values.
+        // not using Metadata.merge() as it will keep both values.
+        if (metadata.get(key).length < 1) {
+          metadata.set(key, value);
+        }
+      }
 
-    this._sendingPromises.push(promise);
-    const popPromise = () => {
-      const index = this._sendingPromises.indexOf(promise);
-      this._sendingPromises.splice(index, 1);
+      return metadata;
     };
-    promise.then(popPromise, popPromise);
-  }
 
-  onInit(config: OTLPGRPCExporterConfigNode): void {
-    // defer to next tick and lazy load to avoid loading grpc too early
-    // and making this impossible to be instrumented
-    setImmediate(() => {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const { onInit } = require('./util');
-      onInit(this, config);
+    this.compression = configureCompression(config.compression);
+    this._transport = new GrpcExporterTransport({
+      address: this.getDefaultUrl(config),
+      compression: this.compression,
+      credentials: credentialProvider,
+      grpcName: grpcName,
+      grpcPath: grpcPath,
+      metadata: metadataProvider,
+      timeoutMillis: this.timeoutMillis,
     });
   }
 
+  onInit() {
+    // Intentionally left empty; nothing to do.
```

Comment:
This is `abstract` in `OTLPExporterBase`, so that's why I did not remove it.

I intend to completely remove this `OTLPExporterBase` at some point in a follow up (though it will likely take a few PRs to get there) by changing the structure of the exporters to be more modular. A proposal on how that new base *could* look like can be seen [here (this is a single commit of my exporter rewrite proposal)](https://github.com/open-telemetry/opentelemetry-js/pull/4415/commits/0b9afdb2b39b0aa29822b0c79a29a3b71d9230bc). There the shared code for all exporters is represented by the `OTLPExportDelegate`. It'll make use of the same transport interfaces I came up with in this PR. 

This PR here is actually split from work that I have done on #4415, which does something similar for `http` based exporters. I had locally written parts that became this PR to prove that the interfaces from #4415 would work with gRPC as well.